### ISSUE TITLE
Properly initialize dBFT context after CVs

### ISSUE THREAD

**Issue by AnnaShaleva** [bane-labs/go-ethereum#54]
**TL;DR**
We're missing a proposal update mechanism for the case of change views. It leads to:
 1) possible inability to process non-empty blocks (blocks with transactions) after first CV, which is critical for the network functioning;
 2) non-changeable proposal in subsequent views of the same height, which is not critical but isn't good.

**Full issue description**
After #49 the pipeline of a single dBFT round is the following: miner submits new sealing task to consensus node. Once new sealing task is received by `DBFT` service, dBFT initialization at the proper height and zero view is being performed. Primary builds PrepareRequest based on its own sealing task and broadcasts it. Backup receives PrepareRequest, verifies it and (if it's valid) rewrites its own sealing task with the one got from PrepareRequest. Starting from this point, all nodes that have received PrepareRequest share the same sealing task. It's OK until the first CV occurs. And after first CV occurs (with a valid proposal of the past view), we have two main problems:
1. Since all nodes (that received and successfully validated PrepareRequest) share the same sealing proposal, PrepareRequest of new view won't differ from PrepareRequest of the previous view. We're OK with some header fields being unchanged, but we're not OK with a set of transactions (which is critical) and block time (which is less critical).
2. Since sealing proposal of backup node was replaced with the sealing proposal of primary after PrepareRequest receiving, the set of sealing transactions is being reset to nil. If proposal from previous view has some sealing transactions, then proposal of the current view won't be properly built by new primary, because it doesn't have sealing transactions, thus, block won't be collected. This issue is just a consequence of the first problem.

**Solution**
Within the scope of this issue we need to refactor integration between `DBFT` engine and miner, so that initialisation of sealing proposal is being performed right after subsequent change view. In other words, what we need to have is: new view -> new sealing proposal from miner (with updated header's fields and updated set of transactions) -> new prepare request for this view initialized by new sealing proposal.
